Vizzuality/heco-invest
# Project developer open calls dashboard w/ sorting + withdrawing an open call

## ~IMPORTANT~

~I've used mocked data, but the correct endpoints.~
~Once #513 is merged we'll make sure things work as expected.~

## Description

This PR adds the _"Open call applications"_ table to the project developer's dashboard.  

#### Implemented
- List of open call applications  
- Sorting  
- Searching  
- View for when there aren't any open call applications yet  
- Link to the open call application details page
  _To be implemented in [LET-899](https://vizzuality.atlassian.net/browse/LET-899)_  
- Confirmation modal to withdraw an open call application  

#### Notes

I moved application-related hooks to `open-calls/applicationService.ts`. I think `open-calls/open-call-service.ts` was getting too bloated and it's getting hard to find things. I envision that in the future we'll have something like: 

```
- services
  - open-calls  
    - index.ts (base CRUD)
    - applications.ts (related to open call applications applications)  
    - favourite.ts (related to favouriting them)
    - account.ts (specific to the account's open calls)
```

I think in the future we should set on a set naming style, possibly [Airbnb's (for no particular reason)](https://github.com/airbnb/javascript/tree/master/react#naming)

## Testing instructions

Verify that a project developer:  
- can see a list of open calls they applied to, in the dashboard  
- the table is sortable  
- it is possible to withdraw an open call application  

## Tracking

[LET-618 | FE - PD users can see open calls to which they applied in the Dashboard](https://vizzuality.atlassian.net/browse/LET-618)
[LET-973 | FE - PD users can sort the open calls they have applied to](https://vizzuality.atlassian.net/browse/LET-973)
[LET-982 | FE - PD users can withdraw a project (sic) application in their dashboard](https://vizzuality.atlassian.net/browse/LET-982)

## Screenshots  

**No open call applications**
<img width="1680" alt="Screenshot 2022-08-29 at 17 01 26" src="https://user-images.githubusercontent.com/6273795/187250434-a41279d5-5ec2-470d-b405-9d627df69e41.png">

**Open call applications list**
<img width="1680" alt="pd-oc-lst" src="https://user-images.githubusercontent.com/6273795/187250465-5d855153-fe63-40d7-8b05-8e77552d964e.png">

**Withdrawing an open call application** 
<img width="1680" alt="Screenshot 2022-08-29 at 17 00 27" src="https://user-images.githubusercontent.com/6273795/187250505-a5e5c00b-5956-4c46-9dda-97ddf8f7ed66.png">

**Endpoint errors**  
<img width="725" alt="Screenshot 2022-08-29 at 17 36 33" src="https://user-images.githubusercontent.com/6273795/187250671-ec4513de-fcdf-4d9b-bfd1-27737dd41641.png">
